src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -464,7 +463,7 @@ private async Task OnPartialValidationSucceededAsync(ChainedHeader chainedHeader
                 }
             }
 
-            if (chainedHeaderBlocksToValidate != null)
+            if (fullValidationRequired && chainedHeaderBlocksToValidate != null)
```

Comment:
This condition should have been extended with check that the full validation succeeded. So, if there was no full validation, partial validation should continue. And if there was full validation, partial validation should continue only if full validation succeeded.

Reply:
Ok thankals @aprogiena. That makes it more clear. I think im going to get FullyValidateAsync to return a result class for better feedback. Let me check what I can do.

Reply:
all you need is 
```
bool nextValidationNeeded = chainedHeaderBlocksToValidate != null;
bool fullValidationFailed = (connectBlocksResult != null) && connectBlocksResult.Succeeded;
if (nextValidationNeeded && !fullValidationFailed)
```
or something like that